### DISCUSSION TITLE
[Assets] Add a download button to the Image Modal

### DISCUSSION THREAD

**Discussion by davidhellmann** [craftcms/cms#13489] [Ideas] [answered]
I need that a lot here and there. The current flow is to copy the name > go to assets > search for that file > download it. 
A `Preview` | `Edit` | `Download` button on each asset would be great. Speed up things a lot for cases like this :)

![CleanShot 2023-07-31 at 12 21 01](https://github.com/craftcms/cms/assets/764292/cf498565-bacf-47ef-aefb-5f076539fca9)

**Comment by thupsi** [comment]
In your screenshot there is the `Open the full edit page in a new tab` button on the top right. The `download` button is accessible on the `full edit page`. 

Not one click, but much faster than searching with the asset name!

**Reply by davidhellmann** [reply]
Sure but then I have also to deal with a new tab. I know, not a big deal at all but I think the button is a easy win. 😅

**Comment by mmikkel** [comment]
I'm all for a download button inside the slideout, but if it happens I hope a better placement can be found in the slideout UI for it, as well as the existing actions (Preview & Edit Image). Never been a big fan of these buttons on top of the thumbnail, that are only visible on mouseover. Adding additional buttons without addressing that first, will probably make it feel even more cramped/awkward :)

**Comment by brandonkelly** [comment]
We’re working on making element actions more consistent between edit pages and slideouts for Craft 5.

**Comment by johnj24** [comment]
I have a client that would love the ability to download PDFs straight from the element slideout.

**Comment by brandonkelly** [comment, marked as answer]
This was made possible in Craft 5, via the new action menu available within asset slideouts.

<img src="https://github.com/craftcms/cms/assets/47792/eedc7f36-9b17-4f45-9a90-010b04ad8145" width="351" height="300" alt="An asset slideout’s expanded action menu, highlighting the “Download” action">
